pkg/server/server.go

```diff
@@ -255,79 +255,79 @@ func (s *ProxyServer) getBackend(reqHost string) (Backend, error) {
 	return nil, &ErrNotFound{}
 }
 
-func (s *ProxyServer) addBackend(agentID string, conn agent.AgentService_ConnectServer) (backend Backend) {
+func (s *ProxyServer) addBackend(agentID string, backend *Backend) {
```

Comment:
Having switched these from an interface being passed to a (pointer to)struct, are we sure we got all the pointer to parts? Otherwise we could have a nasty and subtle bug.

Reply:
Moot point now, but:

One of the nasty and subtle bugs I chased down was that a unit test was still passing agent.AgentService_ConnectServer when it should pass a Backend.

I intentionally added a new method to Backend interface in order to break the interchangeability between the gRPC stream and the abstraction.